### PR TITLE
[webui] Do not show auto-accept when request is in final state

### LINKED ISSUE #3318: Do not show auto-accept timelimit when rq is final
https://build.opensuse.org/request/show/506639

"This request will get accepted in <x> days automatically!" next to the rq status.

This ought not to be shown when the rq is in a final state.

### PR DESCRIPTION
**Pull request by ChrisBr**
Fix #3318.

request_auto_accept

### CODE REVIEW

**Comment by ChrisBr** [comment]
@jengelh @andreasstieger @adrianschroeter please have a look

**Comment by adrianschroeter** [comment]
"The request auto-accept was set to" sounds a bit weird, maybe just "Auto-accept was set to ..." ?

@evanrolfe do you have a better proposal as native speaker?

**Comment by jengelh** [comment]
"Auto-accept will occur on 2017-XX-YY" / "has occurred on 2017-XX-YY".

**Comment by ChrisBr** [comment]
@bgeuken it's only view changes, do you really want a feature test for that?

File: src/api/app/views/webui/request/show.html.erb [hunk 1]
```diff
@@ -50,11 +50,16 @@
         <% end %>
         <% if @accept_at -%>
             <li><%= sprite_tag('exclamation', title: '') %>
-               <% if Time.now > Time.parse(@accept_at.to_s) %>
-                 This request will get accepted automatically when it enters the 'new' state!
+              <% if BsRequest::FINAL_REQUEST_STATES.include?(@state) %>
+                The request auto-accept was set to
```
Comment (evanrolfe) [inline src/api/app/views/webui/request/show.html.erb]: [Nitpick] 

Should be: "The request's auto-accept was set to"

Or @adrianschroeter s suggestion also works:
"Auto-accept was set to ..."

File: src/api/app/views/webui/request/show.html.erb [hunk 2]
```diff
@@ -50,11 +50,16 @@
...
+                <span title='<%= @accept_at.utc.strftime('%Y-%m-%d %H:%M UTC') %>' class='fuzzy-time'>
+                  <%= @accept_at.utc.strftime('%Y-%m-%d') %>.
+                </span>
+              <% elsif @accept_at.past? %>
+                This request will get accepted automatically when it enters the 'new' state!
```
Comment (evanrolfe) [inline src/api/app/views/webui/request/show.html.erb]: [Nitpick]

Slightly better:
"The request will be automatically accepted  when it enters the 'new' state." IMO we don't need an exclamation mark.

File: src/api/app/views/webui/request/show.html.erb [hunk 3]
```diff
@@ -50,11 +50,16 @@
...
               <% else %>
-               This request will get accepted in
-              <%= distance_of_time_in_words(Time.now, Time.parse(@accept_at.to_s)) %> automatically!
+                This request will get accepted in
```
Comment (evanrolfe) [inline src/api/app/views/webui/request/show.html.erb]: [Nitpick]

"The request will be automatically accepted in 5 minutes."

File: src/api/app/views/webui/request/show.html.erb [hunk 4]
```diff
@@ -50,11 +50,16 @@
         <% end %>
         <% if @accept_at -%>
             <li><%= sprite_tag('exclamation', title: '') %>
-               <% if Time.now > Time.parse(@accept_at.to_s) %>
-                 This request will get accepted automatically when it enters the 'new' state!
+              <% if BsRequest::FINAL_REQUEST_STATES.include?(@state) %>
+                The request auto-accept was set to
+                <span title='<%= @accept_at.utc.strftime('%Y-%m-%d %H:%M UTC') %>' class='fuzzy-time'>
```
Comment (evanrolfe) [inline src/api/app/views/webui/request/show.html.erb]: Do we not have a standard datetime format to use? i.e. so we don't have to repeat `%Y-%m-%d %H:%M UTC'` everytime we print a datetime?

File: src/api/app/views/webui/request/show.html.erb [hunk 5]
```diff
@@ -50,11 +50,16 @@
...
+                  <%= @accept_at.utc.strftime('%Y-%m-%d') %>.
```
Comment (evanrolfe) [inline src/api/app/views/webui/request/show.html.erb]: Same here

File: src/api/app/views/webui/request/show.html.erb [hunk 6]
```diff
@@ -50,11 +50,16 @@
...
+                </span>
+              <% elsif @accept_at.past? %>
+                This request will get accepted automatically when it enters the 'new' state!
               <% else %>
-               This request will get accepted in
-              <%= distance_of_time_in_words(Time.now, Time.parse(@accept_at.to_s)) %> automatically!
+                This request will get accepted in
+                <%= distance_of_time_in_words(Time.now, Time.parse(@accept_at.to_s)) %> automatically!
```
Comment (evanrolfe) [inline src/api/app/views/webui/request/show.html.erb]: What kind of object is `@accept_at`? Maybe you can do something like this?
```ruby
distance_of_time_in_words(Time.now, =@accept_at.to_time))
```

**Comment by bgeuken** [comment]
> ChrisBr commented 2 hours ago
> @bgeuken it's only view changes, do you really want a feature test for that?

It seems to be important so we should test it.

File: src/api/app/views/webui/request/show.html.erb (hunk 4 above)
Comment (ChrisBr) [inline src/api/app/views/webui/request/show.html.erb]: Nope! Added now...

**Comment by ChrisBr** [comment]
@bgeuken @evanrolfe updated and added a feature test, please have a look

File: src/api/config/locales/en.yml [hunk 7]
```diff
@@ -20,4 +20,7 @@
 # available at http://guides.rubyonrails.org/i18n.html.
 
 en:
-  hello: "Hello world"
+  time:
+    formats:
+      only_date: '%Y-%m-%d'
+      default: '%Y-%m-%d %H:%M UTC'
```
Comment (evanrolfe) [inline src/api/config/locales/en.yml]: Brilliant! Thanks

**Pull request merged**